v1/execution/miner.py

```diff
@@ -1,44 +1,60 @@
 from hashlib import sha256
 from random import randint
-from socket import socket
+from socket import socket, AF_INET, SOCK_DGRAM
 from typing import Dict, List
 from json import dumps
 from gnupg import GPG
-from utils import LOCALHOST
+from time import time
+from utils import LOCALHOST, Event
 from semantic.block import Block
 
 class Miner:
-    def __init__(self, master_port: int, block: Block, difficulty: int,  parent_hash = None):
+    def __init__(self, master_name: str, master_port: int, block: Block, parent_hash, gpg: GPG):
+        self.master_name = master_name
         self.master_port = master_port
         self.block = block
-        self.difficulty = difficulty
         self.parent_hash = parent_hash
+        self.gpg = gpg
+        self.txs_str = dumps([str(tx) for tx in block.transactions.values()])
 
-    def proof_of_work(self, last_block_hash):
```

Comment:
Eliminé el proof of work porque me parece que el minador no debe encargarse de validar nada. Es el Master el que se encarga de validar todo.